### PR TITLE
Bug D-140

### PR DESCRIPTION
**Pull request by emilioward**
**What is the context of this PR?**

[https://trello.com/c/OcRpay64](https://trello.com/c/OcRpay64)

This PR fixes a bug that was causing foreign key violations constraint when running the modifier tests. This PR adds in the ability to enable foreign pragma for foreign keys and updates the code and tests to run successfully.

**How to review**

Run test_modifier.py and all the tests should pass

### CODE REVIEW

**Comment by emilioward** [comment]
[BUG D-140 Bug del_draft in modifier.py |3|](https://trello.com/c/OcRpay64/379-bug-d-140-bug-deldraft-in-modifierpy-3)

**Comment by emilioward** [comment]
More bugs have cropped as a result of implementing the foreign key pragma in other sections of the code. These are failing the travis build with foreign key violations.

File: tests/app/test_modifier.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from flask import current_app
 from flask import g
 from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+from sqlalchemy import event
```
Comment (dcdarrell9) [inline tests/app/test_modifier.py]: With the imports, you can add them on the same line......  `from sqlalchemy import create_engine, event`

Comment (emilioward) [inline tests/app/test_modifier.py]: Fixed all files related to the PR

File: app/repository/modifier.py [hunk 2]
```diff
@@ -78,38 +78,37 @@ def del_unread(message, user):
...
-                                       thread_id=draft.thread_id, collection_case=draft.collection_case,
-                                       ru_id=draft.ru_id, collection_exercise=draft.collection_exercise,
-                                       survey=draft.survey)
 
         try:
-            session.add(secure_message)
+            session.query(SecureMessage).filter_by(msg_id=draft_id).update({"subject": draft.subject,
+                                                                            "body": draft.body})
             session.commit()
         except Exception as e:
```
Comment (emilioward) [inline app/repository/modifier.py]: These seem to be the same through most the code in other files, might need to be raised as another looked at in the future maybe create a new card?

Comment (rjweeks) [inline app/repository/modifier.py]: I agree using specific exceptions is best practice.

**Pull request merged**